client/virtual_channel_test.go

```diff
@@ -68,6 +68,60 @@ func TestVirtualChannelsOptimistic(t *testing.T) {
 	assert.NoError(t, err, "Bob: invalid final balances")
 }
 
+func TestVirtualChannelsDispute(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), testDuration)
+	defer cancel()
+
+	vct := setupVirtualChannelTest(t, ctx)
+	var err error
+	assert := assert.New(t)
+
+	time.Sleep(100 * time.Millisecond) // Sleep to ensure that virtual channel watcher is running.
+
+	// Alice registers and settles channel with Ingrid.
+	err = vct.registerAndSettle(ctx, vct.chAliceIngrid)
+	assert.NoError(err, "alice-ingrid: register and settle")
+
+	// Ingrid registers and settles channel with Alice.
+	err = vct.registerAndSettle(ctx, vct.chIngridAlice)
+	assert.NoError(err, "ingrid-alice: register and settle")
+
+	// Bob registers and settles channel with Ingrid.
+	err = vct.registerAndSettle(ctx, vct.chBobIngrid)
+	assert.NoError(err, "bob-ingrid: register and settle")
+
+	// Ingrid registers and settles channel with Bob.
+	err = vct.registerAndSettle(ctx, vct.chIngridBob)
+	assert.NoError(err, "ingrid-bob: register and settle")
```

Comment:
As we talked about before, it would be ideal to assert the on-chain balances afterwards.

Reply:
OK, I will try to do that if it's not too much effort.

Reply:
Turns out it's quite a bit of work. Are you okay with doing that in a follow-up PR?

Reply:
Yes sure, it was not in there before as well.

Reply:
OK, took some lines of code but was feasible.

Reply:
Very nice!